src/libstore/globals.hh

```diff
@@ -142,6 +142,30 @@ public:
         )",
         {"build-cores"}, false};
 
+    Setting<unsigned int> loadLimit{
+      this,
+      getDefaultCores(),
+      "load-limit",
+      R"(
+        Sets the value of the `NIX_LOAD_LIMIT` environment variable in the
+        invocation of builders. Builders can use this value at their discretion
+        to dynamically control the amount of parallelism with respect to the
+        machine's load average. For instance, in Nixpkgs, if this value is >0,
+        the builder passes the `-lN` flag to GNU Make. In this case, if the
+        load average of the machine exceeds `N`, the amount of parallelism will
+        be dynamically reduced to 1.
+
+        By default, it is set to the number of cores available to Nix.
+
+        On busy machines where Nix co-exists with other workloads, parallelism
+        may not work as intended. For example, consider a 64 core machine whose
+        load average is 24 and where Nix is limited to 8 cores. By default,
+        `-j8 -l8` will be passed to GNU Make. Since the load average exceeds 8,
+        no parallelism will take place despite the fact that 8 cores are
+        available. In this case, `load-limit` should be set to `0` to prevent
+        the `-lN` flag from being passed to GNU Make.
```

Comment:
```suggestion
        available. In this case, `load-limit` should be set to `-1`, which GNU Make would interpret as "no limit".
```

Setting it to zero won't necessarily prevent it to be passed to Make, that would depend on how a builder is implemented. Right now [`-l` is not set at all in Nixpkgs builders](https://github.com/NixOS/nixpkgs/pull/192447).
Make's source code says that the [`-1` (default) means no limit](https://github.com/wkusnierczyk/make/blob/c93ec39d1079224f28ebb5797243d38266a1d066/main.c#L273-L283).